### ISSUE TITLE
Cassandra 4 support

### ISSUE THREAD

**Issue by DennisBB** [JanusGraph/janusgraph#2325]
When Cassandra 4 is gonna be supported?

**Comment by porunov** [comment]
There are no stable Cassandra 4 releases yet. Right now the latest release is `cassandra-4.0-beta4`.
That said, I didn't check Cassandra 4 beta compatibility with JanusGraph yet.

**Comment by DennisBB** [comment]
> Apache Cassandra 3.0 is supported until 6 months after 4.0 release (date TBD)

I think it's better to start preparing till it's out

**Comment by gklijs** [comment]
Release candidate is out now, just fyi, not using JanusGraph at the moment.

**Comment by DennisBB** [comment]
Cassandra 4 has just been released that supports Java 11

Can JanusGraph 0.6.0 be released with Cassandra 4 support?

It would make sense since both should be able to run on Java 11

**Comment by porunov** [comment]
@farodin91 @FlorianHockmann do you think it's possible to upgrade Cassandra to 4.x knowing that `Thrift` was removed from 4.x version but we use it in `CqlBinaryInputFormat` for Hadoop?
```
import org.apache.cassandra.thrift.SlicePredicate;
import org.apache.cassandra.thrift.SliceRange;
```

**Comment by FlorianHockmann** [comment]
It should work without Thrift. @pluradj tried that out with our PR #1436 with Cassandra that had Thrift disabled completely. So, I think that this should also work with Cassandra 4, but we should definitely test that. I guess these types were still in the Thrift package but we don't actually use the Thrift protocol only because we use these types.

We will however need to replace these types with something else when we update our `cassandra.version` to 4 as the `thrift` package was removed completely in that version.

**Comment by DennisBB** [comment]
Can Cassandra 4 also be supported in 0.6.0?

**Comment by farodin91** [comment]
Closed by  #3033
